packages/pkcs12/src/index.ts

```diff
@@ -0,0 +1,106 @@
+import crypto, { HexBase64Latin1Encoding, Utf8AsciiLatin1Encoding } from 'crypto';
+import forge from 'node-forge';
+
+/**
+ * returns serial number as an uppercased hexidecimal string
+ */
+export function getFormattedSerialNumber(certificate: forge.pki.Certificate): string | null {
+  const { serialNumber } = certificate;
+  return serialNumber ? serialNumber.replace(/^0+/, '').toUpperCase() : null;
+}
+
+/**
+ * Extracts a certificate from PKCS#12
+ * This is assumed to be a conventional PKCS#12 where there is exactly one certificate and one key
+ */
+export function getX509Certificate(p12: forge.pkcs12.Pkcs12Pfx): forge.pki.Certificate {
+  const certBagType = forge.pki.oids.certBag;
+  const bags = p12.getBags({ bagType: certBagType })[certBagType];
+  if (!bags || bags.length === 0) {
+    throw new Error(`PKCS12: No certificates found`);
+  }
+  const certificate = bags[0].cert;
+  if (!certificate) {
+    throw new Error('PKCS12: bag is not a certificate');
+  }
+  return certificate;
+}
+
+/**
+ * Extracts a certificate from PKCS#12
+ * This is assumed to be a PKCS#12 in Keystore format where the friendlyName (alias) contains a PrivateKeyEntry
+ * A PrivateKeyEntry contains exactly one certificate and one key
+ * https://docs.oracle.com/javase/7/docs/api/java/security/KeyStore.PrivateKeyEntry.html
+ */
+export function getX509CertificateByFriendlyName(
+  p12: forge.pkcs12.Pkcs12Pfx,
+  friendlyName: string
+): forge.pki.Certificate {
+  const certBagType = forge.pki.oids.certBag;
+  const bags = p12.getBags({ friendlyName, bagType: certBagType }).friendlyName;
+  if (!bags || bags.length === 0) {
+    throw new Error(`PKCS12: No certificates found under friendlyName: ${friendlyName}`);
+  }
+  const certificate = bags[0].cert;
+  if (!certificate) {
+    throw new Error('PKCS12: bag is not a certificate');
+  }
+  return certificate;
+}
+
+export function getPKCS12(
```

Comment:
maybe `parse` or `decode` would be better here

Reply:
Decrypt?

Reply:
I'm not sure `pkcs12FromAsn1` returns decrypted values, but if that is the case then yes decrypt would be a better name.

Reply:
i think parse would be better here -- it definitely parse/decodes the base64 p12 to an object. It also optionally uses the password to decrypt the p12 if the content type is marked as 'encryptedData' https://github.com/digitalbazaar/forge/blob/588c41062d9a13f8dc91be3723b159c6cc434b15/lib/pkcs12.js#L554